docs/upgrade/troubleshooting.md

```diff
@@ -0,0 +1,126 @@
+# Troubleshooting
+
+## Overview
+
+Here are some tips to troubleshoot a failed upgrade:
+
+- Check [version-specific upgrade notes](./automatic.md#upgrade-support-matrix). You can click the version in the support matrix table to see if there are any known issues.
+- Dive into the upgrade [design proposal](https://github.com/harvester/harvester/blob/master/enhancements/20220413-zero-downtime-upgrade.md). The following section briefly describes phases within an upgrade and possible diagnostic methods.
+
+## Diagnose the upgrade flow 
+
+A Harvester upgrade process contains several phases. 
+    ![](./assets/ts_upgrade_phases.png)
+
+### Phase 1: Provision upgrade repository VM.
+
+The Harvester controller downloads a Harvester release ISO file and uses it to provision a VM. During this phase you can see the upgrade status windows show:
+
+![](./assets/ts_status_phase1.png){: style="width:60%"}
+
+The time to complete the phase depends on the user's network speed and cluster resource utilization. We see failures in this phase due to network speed. If this happens, the user can [start over the upgrade](#start-over-an-upgrade) again.
+
+We can also check the repository VM (named with the format `upgrade-repo-hvst-xxxx`) status and its corresponding pod:
+
+```
+$ kubectl get vm -n harvester-system
+NAME                              AGE    STATUS     READY
+upgrade-repo-hvst-upgrade-9gmg2   101s   Starting   False
+
+$ kubectl get pods -n harvester-system | grep upgrade-repo-hvst
+virt-launcher-upgrade-repo-hvst-upgrade-9gmg2-4mnmq     1/1     Running     0          4m44s
+```
+
+### Phase 2: Preload container images
+
+The Harvester controller creates jobs on each Harvester node to download images from the repository VM and preload them. These are container images required for the next release.
+
+During this stage you can see the upgrade status windows shows:
+
+![](./assets/ts_status_phase2.png){: style="width:60%"}
+
+It will take a while for all nodes to preload images. If the upgrade fails at this phase, the user can check job logs in `cattle-system` namespace:
```

Comment:
```suggestion
It will take a while for all nodes to preload images. If the upgrade fails at this phase, the user can check job logs in the `cattle-system` namespace:
```